packages/iov-core/src/jsonrpcsigningserver.spec.ts

```diff
@@ -0,0 +1,260 @@
+  // ganache second identity
+  await profile.createIdentity(secp256k1Wallet.id, ethereumConnection.chainId(), HdPaths.bip44(60, 0, 0, 1));
+
+  const core = new SigningServerCore(
+    profile,
+    signer,
+    defaultGetIdentitiesCallback,
+    defaultSignAndPostCallback,
+  );
+  return new JsonRpcSigningServer(core);
+}
+
+import { JsonRpcSigningServer } from "./jsonrpcsigningserver";
+import { TransactionEncoder } from "./transactionencoder";
+
+describe("JsonRpcSigningServer", () => {
+  const ganacheSecondIdentity: PublicIdentity = {
+    chainId: ethereumChainId,
+    pubkey: {
+      algo: Algorithm.Secp256k1,
+      data: Encoding.fromHex(
+        "041d4c015b00cbd914e280b871d3c6ae2a047ca650d3ecea4b5246bb3036d4d74960b7feb09068164d2b82f1c7df9e95839b29ae38e90d60578b2318a54e108cf8",
+      ) as PublicKeyBytes,
+    },
+  };
+
+  const defaultAmount: Amount = {
+    quantity: "1",
+    fractionalDigits: 9,
+    tokenTicker: "CASH" as TokenTicker,
+  };
+
+  it("can get bnsd identities", async () => {
+    pendingWithoutBnsd();
+    pendingWithoutEthereum();
+
+    const bnsConnection = await bnsConnector(bnsdUrl).client();
+
+    const server = await makeBnsEthereumSigningServer();
+
+    const response = await server.handleUnchecked({
+      jsonrpc: "2.0",
+      id: 123,
+      method: "getIdentities",
+      params: {
+        reason: "Who are you?",
+        chainIds: [bnsConnection.chainId()],
+      },
+    });
+    expect(response.id).toEqual(123);
+    if (isJsonRpcErrorResponse(response)) {
+      throw new Error(`Response must not be an error, but got '${response.error.message}'`);
+    }
+    const result = TransactionEncoder.fromJson(response.result);
+    expect(result).toEqual(jasmine.any(Array));
+    expect((result as ReadonlyArray<any>).length).toEqual(1);
+    expect(result[0].chainId).toEqual(bnsConnection.chainId());
+    expect(result[0].pubkey).toEqual({
+      algo: Algorithm.Ed25519,
+      data: fromHex("533e376559fa551130e721735af5e7c9fcd8869ddd54519ee779fce5984d7898"),
+    });
+
+    server.shutdown();
+    bnsConnection.disconnect();
+  });
+
+  it("can get ethereum identities", async () => {
+    pendingWithoutBnsd();
+    pendingWithoutEthereum();
+
+    const server = await makeBnsEthereumSigningServer();
+
+    const response = await server.handleChecked({
+      jsonrpc: "2.0",
+      id: 123,
+      method: "getIdentities",
+      params: {
+        reason: "Who are you?",
+        chainIds: [ethereumChainId],
+      },
+    });
+    expect(response.id).toEqual(123);
+    if (isJsonRpcErrorResponse(response)) {
+      throw new Error(`Response must not be an error, but got '${response.error.message}'`);
+    }
+    const result = TransactionEncoder.fromJson(response.result);
+    expect(result).toEqual(jasmine.any(Array));
+    expect((result as ReadonlyArray<any>).length).toEqual(1);
+    expect(result[0]).toEqual(ganacheSecondIdentity);
+
+    server.shutdown();
+  });
```

Comment:
In this test there is no Ethereum testing connection (just the one inside of the server that is disconnected during shutdown). The testing connection in the test above only exists because the chain ID in weave local test chains is generated randomly. In Ethereum tests, we use a static chain ID.